modules/FvwmMFL/FvwmMFL.c

```diff
@@ -623,23 +628,52 @@ static void
 fvwm_read(int efd, short ev, void *data)
 {
 	FvwmPacket	*packet;
+	char *sock_path;
+
+	sock_path = set_socket_path();
 
 	if ((packet = ReadFvwmPacket(efd)) == NULL) {
 		if (debug)
 			fprintf(stderr, "Couldn't read from FVWM - exiting.\n");
-		unlink(SOCK);
+		unlink(sock_path);
 		exit(0);
 	}
 
 	broadcast_to_client(packet);
 }
 
+static char *set_socket_path()
+{
+	char *mflsock_env, *sock_path;
+
+	/* Figure out if we are using default MFL socket path or we should respect */
+	/* environment variable FVWMMFL_SOCKET for FvwmMFL socket path */
+	xasprintf(&sock_path, "%s", MFL_SOCKET_DEFAULT);
+
+	mflsock_env = getenv("FVWMMFL_SOCKET");
+	if (mflsock_env != NULL) {
+		const char      *unrolled_path;
+
+		unrolled_path = expand_path(mflsock_env);
+
+		if (unrolled_path[0] == '/')
+			sock_path = fxstrdup(unrolled_path);
+		else
+			xasprintf(&sock_path, "%s/%s", getenv("FVWM_USERDIR"), unrolled_path);
+
+		free((char *)unrolled_path);
+	}
+	return (sock_path);
+}
+
 int main(int argc, char **argv)
 {
 	struct event_base     *base;
 	struct evconnlistener *fmd_cfd;
 	struct sockaddr_un    sin;
 
+	char *sock_path;
```

Comment:
Suggest making this a static variable at the top of this file, outside of `main()`